engine/engine.go

```diff
@@ -437,12 +436,139 @@ func (e *Engine) MakeRangeQueryFromPlan(ctx context.Context, q storage.Queryable
 
 // NewInstantQuery implements the promql.Engine interface.
 func (e *Engine) NewInstantQuery(ctx context.Context, q storage.Queryable, opts promql.QueryOpts, qs string, ts time.Time) (promql.Query, error) {
-	return e.MakeInstantQuery(ctx, q, fromPromQLOpts(opts), qs, ts)
+	return e.MakeInstantQuery(ctx, q, FromPromQLOpts(opts), qs, ts)
 }
 
 // NewRangeQuery implements the promql.Engine interface.
 func (e *Engine) NewRangeQuery(ctx context.Context, q storage.Queryable, opts promql.QueryOpts, qs string, start, end time.Time, step time.Duration) (promql.Query, error) {
-	return e.MakeRangeQuery(ctx, q, fromPromQLOpts(opts), qs, start, end, step)
+	return e.MakeRangeQuery(ctx, q, FromPromQLOpts(opts), qs, start, end, step)
+}
+
+// Distributed constructors
+
+// MakeDistributedInstantQuery creates an instant query that is distributed among the passed endpoints.
+func (e *Engine) MakeDistributedInstantQuery(ctx context.Context, q storage.Queryable, endpoints api.RemoteEndpoints, opts *QueryOpts, qs string, ts time.Time) (promql.Query, error) {
```

Comment:
q is actually used by distriubted engine sometimes - it feels like a bug, but fixing is out of scope of this PR.